xcdat/bounds.py

```diff
@@ -289,16 +348,7 @@ def _add_bounds(self, axis: CFAxisName, width: float = 0.5) -> xr.Dataset:
 
         .. [2] https://cf-xarray.readthedocs.io/en/latest/generated/xarray.Dataset.cf.add_bounds.html#
         """
-        # Add coordinate bounds to the dataset
-        ds = self._dataset.copy()
-        coord_var: xr.DataArray = get_axis_coord(ds, axis)
-
-        # Validate coordinate shape and dimensions
-        if coord_var.ndim != 1:
-            raise ValueError(
-                f"Cannot generate bounds for coordinate variable '{coord_var.name}'"
-                " because it is multidimensional coordinates."
-            )
+        # Validate coordinate shape.
         if coord_var.shape[0] <= 1:
             raise ValueError(
                 f"Cannot generate bounds for coordinate variable '{coord_var.name}'"
```

Comment:
This check is the root of our issues with singleton coordinate variables.

We are trying to index on a 0-dimensional array, which is not possible with `height`.

```python
Traceback (most recent call last):
  File "<string>", line 1, in <module>
IndexError: too many indices for array: array is 0-dimensional, but 1 were indexed
```

We can create 0-dimensional singleton array by doing this:
```python
ds.coords["height"] = float(1.0)
print(ds.height)

<xarray.DataArray 'height' ()>
array(1.)
Coordinates:
    height   float64 1.0
```

This is not the same as a 1-dimension singleton array:

```python
ds.coords["height"] = np.array([1.0])
print(ds.height)

<xarray.DataArray 'height' (height: 1)>
array([1.])
Coordinates:
  * height   (height) float64 1.0
```

I think we just need to fix this check to take into account 0-dimension singletons too.